security/pkg/cmd/probe_test.go

```diff
@@ -15,9 +15,26 @@
 package cmd
 
 import (
+	"bytes"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
 	"testing"
 )
 
+func executeCommand(root *cobra.Command, args ...string) (output string, err error) {
+	_, output, err = executeCommandC(root, args...)
+	return output, err
+}
+
+func executeCommandC(root *cobra.Command, args ...string) (c *cobra.Command, output string, err error) {
+	buf := new(bytes.Buffer)
+	root.SetOutput(buf)
+	root.SetArgs(args)
+	c, err = root.ExecuteC()
```

Comment:
checked that no flush is needed?
shouldn’t the probe have some output?
how do you check the exit status?

Reply:
shared.Fatalf() calls os.Exit(-1) internally and this is not capture able.
but the test code got this situation, the test will be failed.

I tested it and this is the output.

```
$ go test istio.io/istio/security/pkg/cmd
probe-path or interval are not valid

Usage:
   probe [flags]

Flags:
  -h, --help                          help for probe
      --interval duration             Duration used for checking the target file's last modified time.
      --log_as_json                   Whether to format output as JSON or in plain console-friendly format
      --log_callers                   Include caller information, useful for debugging
      --log_output_level string       The minimum logging level of messages to output, can be one of "debug", "info", "warn", "error", or "none" (default "info")
      --log_rotate string             The path for the optional rotating log file
      --log_rotate_max_age int        The maximum age in days of a log file beyond which the file is rotated (0 indicates no limit) (default 30)
      --log_rotate_max_backups int    The maximum number of log file backups to keep before older files are deleted (0 indicates no limit) (default 1000)
      --log_rotate_max_size int       The maximum size in megabytes of a log file beyond which the file is rotated (default 104857600)
      --log_stacktrace_level string   The minimum logging level at which stack traces are captured, can be one of "debug", "info", "warn", "error", or "none" (default "none")
      --log_target stringArray        The set of paths where to output the log. This can be any path as well as the special values stdout and stderr (default [stdout])
      --probe-path string             Path of the file for checking the availability.

FAIL    istio.io/istio/security/pkg/cmd 0.044s


```